test/membrane_http_adaptive_stream/integration_test/sink_bin_integration_test.exs

```diff
@@ -0,0 +1,169 @@
+defmodule Membrane.HTTPAdaptiveStream.SinkBinIntegrationTest do
+  use ExUnit.Case, async: true
+
+  import Membrane.Testing.Assertions
+  alias Membrane.Testing
+
+  @doc """
+  The boolean flag below controls whether reference HLS content in fixtures directory will be created simultaneously with test content.
+  It should be set only when developing new HLS features that are expected to introduce changes to reference HLS files. Nevertheless it should
+  be done only locally to create and push new reference HLS files and this flag must not be set in remote repository. There is unit test in code below
+  that will cause CI to fail if this flag happens to be set on remote repository. Every new version of reference HSL content must
+  be manually verified by its creator by using some player e.g. ffplay command.
+  """
+  @create_fixtures false
```

Comment:
seems like it should be config's variable and rereferenced here by `Application.compile_env`

Reply:
Or should be put inside `runtime.exs` and set via environmental variable when running locally

Reply:
The purpose of this flag is really simple, and it will probably stay unchanged until someone introduces new features to HLS. I don't see the clear benefits of hiding that variable in `Application.compile_env` or `runtime.exs`. 
Currently, the development flow will look as follows:
* someone adds new features to HLS
* integration test is failing
* in the same file one reads the documentation of the test and changes flag to true
* playlist is being regenerated
* manual check
* setup flag back to false (otherwise test is failing thanks to the assertion on this flag)
The only benefit of moving that flag to `compile_env` is removing this weird assertion on the module attribute, but is it that bad?

Reply:
Makes sense, but what does the tests check if the pipeline we are testing is generating the reference files beforehand? It will mean the tests will never fail. Correct me if I am missing something here 🤔

Reply:
The role of this test is mostly to ensure that HLS is working after introducing non-breaking changes (changes that should not influence produces playlist, e.g code cleanup). Look through the steps I've described once again. Someone generates the files -> manually verifies if the files are correct (e.g. using ffplay), and yes, when files playbacks correctly then the test should not fail (which is expected behaviour). This test mostly proves that HLS is working correctly for some real-life input, and it remains working correctly after introducing changes.

Reply:
Okay, it now makes a lot of sense.